dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala

```diff
@@ -64,40 +70,66 @@ case class Table(
   def put(hashPK: Any, attributes: SimplePk*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, attributes: _*)
   def put(hashPK: Any, rangePK: Any, attributes: SimplePk*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, rangePK, attributes: _*)
 
-  def putItem[E <: AnyRef](entity: E)(implicit dynamoDB: DynamoDB): Unit = {
-    val attrs = getAttrValuesToMap(entity, true)
-
-    if (attrs("keys").exists(x => x._1 == "hashPK") && attrs("keys").exists(x => x._1 == "rangePK")) {
-      val hashPK = attrs("keys").find(f => f._1 == "hashPK").get._2
-      val rangePK = attrs("keys").find(f => f._1 == "rangePK").get._2
-      dynamoDB.put(this, hashPK, rangePK, attrs("attributes"): _*)
-    } else if (attrs("keys").exists(x => x._1 == "hashPK")) {
-      val hashPK = attrs("keys").find(f => f._1 == "hashPK").get._2
-      dynamoDB.put(this, hashPK, attrs("attributes"): _*)
+  def putItem[T: u.TypeTag](entity: T)(implicit dynamoDB: DynamoDB): Unit = {
+    val annotations: List[AnnotationMeta] = getterAnnotationsFromEntity(entity)
+    val fields: Seq[(String, AnyRef)] = getterNamesFromEntity(entity)
+
+    val annotationFields:List[Option[AnnotationMeta]] = annotations.map(a => {
+      val found: Option[AnyRef] = fields
+        .find { case (fieldName, _) => fieldName == a.name }
+        .flatMap{ case(_, getterValue) => Some(getterValue) }
+
+      found match {
```

Comment:
if you want to do this, using `map` method rather than match expression can be simpler.

```
found.map { getterValue => /* the same code */ } // returns Option

Reply:
hey @seratch  - thanks for the review. I guess map seems to work better in this case. Pushed into git. Please take a look